File: src/main/java/com/imcode/imcms/mapping/dao/SystemDao.scala
package com.imcode.imcms.mapping.dao

import com.imcode._
import com.imcode.imcms.mapping.orm.SystemProperty
import org.springframework.transaction.annotation.Transactional
import com.imcode.imcms.dao.hibernate.HibernateSupport


@Transactional(rollbackFor = Array(classOf[Throwable]))
class SystemDao extends HibernateSupport {

  def getProperties(): JList[SystemProperty] = hibernate.listAll()

  def getProperty(name: String): SystemProperty = hibernate.getByQuery(
    "SELECT p FROM SystemProperty p WHERE p.name = ?1", 1 -> name)

  def saveProperty(property: SystemProperty) = hibernate.saveOrUpdate(property)
}

File: src/main/java/com/imcode/imcms/mapping/dao/SystemDao.java
package com.imcode.imcms.mapping.dao;

import com.imcode.imcms.mapping.orm.SystemProperty;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import java.util.List;


@Transactional
public class SystemDao {

    @PersistenceContext
    private EntityManager entityManager;

    public List<SystemProperty> getProperties() {
        return entityManager.createQuery("SELECT p FROM SystemProperty p", SystemProperty.class).getResultList();
    }

    public SystemProperty getProperty(String name) {
        return entityManager.createQuery("SELECT p FROM SystemProperty p WHERE p.name = ?1", SystemProperty.class)
                .setParameter(1, name)
                .getSingleResult();
    }

    public SystemProperty saveProperty(SystemProperty property) {
        return entityManager.merge(property);
    }
}





